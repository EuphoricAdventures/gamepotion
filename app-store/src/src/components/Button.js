import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { withRouter } from 'react-router-dom'

import { font, colours } from '../styleAbstractions'

const StyledButton = styled.button`
  display: block;
  padding: 0.4rem 0.8rem 0.4rem 0.8rem;
  transition: background-color 0.2s ease-in-out, border-color 0.2s ease-in-out, opacity 0.2s ease-in-out;
  ${font}
  border-radius: 4px;
  border-width: 2px;
  border-style: solid;
  outline: 0;
  &.flavour-strong {
    background-color: #2e3131;
    color: ${colours.foreNegative};
    border-color: #2e3131;
  }
  &.flavour-strong:not([disabled]):hover {
    background-color: #6c7a89;
    border-color: #6c7a89;
  }
  &.flavour-weak {
    background-color: transparent;
    color: ${colours.fore};
    border-color: transparent;
  }
  &[disabled] {
    opacity: 0.5;
  }
  :focus {
    border-color: ${colours.outline};
  }
  > img {
    display: block;
    width: 2rem;
    height: 2rem;
    margin: 0 auto 0 auto;
    // background-color: blue;
  }
  > span {
    display: block;
    height: 1.5rem;
    line-height: 1.5rem;
    // background-color: red;
  }
  > img + span {
    margin-top: 0.5rem;
  }
`

const handleOnClick = (history, route, e, onClick) => {
  if (typeof route === 'string') {
    history.push(route)
  } else if (typeof onClick === 'function') {
    return onClick(e)
  }
}

const Button = ({ history, flavour, route, onClick, icon, hint, disabled, children }) => {
  return (
    <StyledButton disabled={disabled} title={hint} onClick={(e) => handleOnClick(history, route, e, onClick)} className={`component--button flavour-${flavour}`}>
      {icon && <img src={icon} alt={hint} />}
      {children && <span>{children}</span>}
    </StyledButton>
  )
}

Button.propTypes = {
  flavour: PropTypes.string,
  route: PropTypes.string,
  onClick: PropTypes.func,
  icon: PropTypes.string,
  hint: PropTypes.string,
  disabled: PropTypes.bool
}

Button.defaultProps = {
  flavour: 'strong',
  onClick: null,
  hint: '',
  disabled: false
}

export default withRouter(Button)
