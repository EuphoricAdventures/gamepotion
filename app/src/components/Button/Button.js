import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { font, outline } from '../abstractions'
import { withRouter } from 'react-router-dom'

const StyledButton = styled.button`
  display: block;
  padding: 0.75rem;
  transition: background-color 0.1s ease-in-out;
  ${font}
  background-color: rgb(48, 48, 48);
  color: white;
  border-radius: 4px;
  :focus {
    ${outline}
  }
  :hover {
    background-color: rgb(64, 64, 64);
  }
`

const Button = ({ history, route, onClick, icon, hint, children }) => {
  const handleOnClick = () => {
    if (typeof route === 'string') {
      history.push(route)
    } else if (typeof onClick === 'function') {
      return onClick()
    }
  }
  return (
    <StyledButton title={hint} onClick={handleOnClick} className='component--button'>
      {children}
      {icon && <img src={icon} alt={hint} />}
    </StyledButton>
  )
}

Button.propTypes = {
  route: PropTypes.string,
  onClick: PropTypes.func,
  icon: PropTypes.string,
  hint: PropTypes.string.isRequired
}

Button.defaultProps = {
  onClick: null
}

export default withRouter(Button)
