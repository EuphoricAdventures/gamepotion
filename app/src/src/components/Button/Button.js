import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { withRouter } from 'react-router-dom'

import { font, colours } from '../../styleAbstractions'

const StyledButton = styled.button`
  display: block;
  padding: 0.4rem 0.8rem 0.4rem 0.8rem;
  transition: background-color 0.2s ease-in-out, border-color 0.2s ease-in-out, opacity 0.2s ease-in-out;
  ${font}
  background-color: #2e3131;
  color: ${colours.foreNegative};
  border-radius: 4px;
  border-width: 2px;
  border-style: solid;
  border-color: #2e3131;
  outline: 0;
  &[disabled] {
    opacity: 0.5;
  }
  :focus {
    border-color: ${colours.outline};
  }
  &:not([disabled]):hover {
    background-color: #6c7a89;
    border-color: #6c7a89;
  }
  > img {
    display: block;
    width: 2rem;
    height: 2rem;
    margin: 0 auto 0 auto;
    // background-color: blue;
  }
  > span {
    display: block;
    height: 1.5rem;
    line-height: 1.5rem;
    // background-color: red;
  }
  > img + span {
    margin-top: 0.5rem;
  }
`

const handleOnClick = (history, route, onClick) => {
  if (typeof route === 'string') {
    history.push(route)
  } else if (typeof onClick === 'function') {
    return onClick()
  }
}

const Button = ({ history, route, onClick, icon, hint, disabled, children }) => {
  return (
    <StyledButton disabled={disabled} title={hint} onClick={() => handleOnClick(history, route, onClick)} className='component--button'>
      {icon && <img src={icon} alt={hint} />}
      {children && <span>{children}</span>}
    </StyledButton>
  )
}

Button.propTypes = {
  route: PropTypes.string,
  onClick: PropTypes.func,
  icon: PropTypes.string,
  hint: PropTypes.string,
  disabled: PropTypes.bool
}

Button.defaultProps = {
  onClick: null,
  hint: '',
  disabled: false
}

export default withRouter(Button)
