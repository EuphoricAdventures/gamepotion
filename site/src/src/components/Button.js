import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import {Link} from 'gatsby'

const StyledButton = styled.div`
  a {
    display: inline-block;
    padding: 0.4rem 0.8rem 0.4rem 0.8rem;
    transition: background-color 0.2s ease-in-out, border-color 0.2s ease-in-out, opacity 0.2s ease-in-out;
    border-radius: 4px;
    border-width: 2px;
    border-style: solid;
    outline: 0;
    color: inherit;
    text-decoration: none;
    > span {
      display: block;
      height: 1.5rem;
      line-height: 1.5rem;
      // background-color: red;
    }
  }
  a.flavour-strong {
    background-color: #2e3131;
    color: #ffffff;
    border-color: #2e3131;
  }
  a.flavour-strong:not([disabled]):hover {
    background-color: #6c7a89;
    border-color: #6c7a89;
  }
  a.flavour-weak {
    background-color: transparent;
    color: #2e3131;
    border-color: transparent;
  }
  a[disabled] {
    opacity: 0.5;
  }
  a:focus {
    border-color: #38EF7D;
  }
`

const Button = ({ flavour, route, hint, disabled, children }) => {

  const isRouteExternal = (route.indexOf('http://') === 0 || route.indexOf('https://') === 0)

  const props = {
    disabled,
    title: hint,
    className: `flavour-${flavour}`
  }

  return (
    <StyledButton className='component--button'>
      {isRouteExternal && <a href={route} {...props}><span>{children}</span></a>}
      {!isRouteExternal && <Link to={route} {...props}><span>{children}</span></Link>}
    </StyledButton>
  )
}

Button.propTypes = {
  flavour: PropTypes.string,
  route: PropTypes.string,
  hint: PropTypes.string,
  disabled: PropTypes.bool
}

Button.defaultProps = {
  flavour: 'strong',
  hint: '',
  disabled: false
}

export default Button
