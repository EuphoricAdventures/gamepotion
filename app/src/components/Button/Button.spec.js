import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import 'jest-styled-components'
import Button from './Button'

it('renders', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Button>Do Something</Button>
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
