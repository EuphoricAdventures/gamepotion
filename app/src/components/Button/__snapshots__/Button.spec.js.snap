// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  display: block;
  padding: 0.7rem;
  -webkit-transition: background-color 0.2s ease-in-out,border-color 0.2s ease-in-out,opacity 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,border-color 0.2s ease-in-out,opacity 0.2s ease-in-out;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",sans-serif;
  font-weight: 400;
  -webkit-text-size-adjust: 100%;
  text-size-adjust: 100%;
  -webkit-font-smoothing: antialiased;
  background-color: #2e3131;
  color: #ffffff;
  border-radius: 4px;
  border-width: 2px;
  border-style: solid;
  border-color: #2e3131;
  outline: 0;
}

.c0[disabled] {
  opacity: 0.5;
}

.c0:focus {
  border-color: #38EF7D;
}

.c0:not([disabled]):hover {
  background-color: #6c7a89;
  border-color: #6c7a89;
}

<button
  className="component--button c0"
  disabled={false}
  onClick={[Function]}
  title="Do Something"
>
  Do Something
</button>
`;
